File: spring-integration/http-integration/inbound-channel-adapter/src/main/java/com/ddubson/integration/IntegrationConfig.java
package com.ddubson.integration;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.integration.dsl.IntegrationFlow;
import org.springframework.integration.dsl.IntegrationFlows;
import org.springframework.integration.dsl.channel.MessageChannels;
import org.springframework.integration.dsl.http.Http;
import org.springframework.integration.http.inbound.HttpRequestHandlingMessagingGateway;
import org.springframework.messaging.Message;
import org.springframework.messaging.MessageChannel;

@Configuration
public class IntegrationConfig {
    private MessagePrinter messagePrinter;

    public IntegrationConfig(MessagePrinter messagePrinter) {
        this.messagePrinter = messagePrinter;
    }

    @Bean("httpChannel")
    public MessageChannel httpChannel() {
        return MessageChannels.direct().get();
    }

    @Bean
    public HttpRequestHandlingMessagingGateway inboundHttpChannelAdapter() {
        return Http.inboundChannelAdapter("/")
                .requestChannel("httpChannel")
                .get();
    }

    @Bean
    public IntegrationFlow integrationFlow() {
        return IntegrationFlows.from(inboundHttpChannelAdapter())
                .<Message<?>, Void>handle(messagePrinter).get();
    }
}

File: spring-integration/routing-messages/payload-type-router/src/main/kotlin/com/ddubson/integration/IntegrationConfig.kt
package com.ddubson.integration

import com.ddubson.CommandLineLogAdapter
import com.ddubson.LogAdapter
import com.ddubson.integration.services.NumericPrintService
import com.ddubson.integration.services.PrintService
import com.ddubson.integration.services.StringPrintService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.integration.annotation.ServiceActivator
import org.springframework.integration.channel.DirectChannel
import org.springframework.integration.dsl.IntegrationFlow
import org.springframework.integration.dsl.IntegrationFlows
import org.springframework.messaging.MessageChannel

@Configuration
class IntegrationConfig {
    @Bean
    fun integrationFlow(): IntegrationFlow {
        return IntegrationFlows.from("inputChannel")
                .route<Any, String>({ i ->
                    when (i) {
                        is Int -> "int"
                        is String -> "string"
                        else -> "def"
                    }
                }, { i ->
                    i.channelMapping("int", "intChannel")
                            .channelMapping("string", "stringChannel")
                }).get()
    }

    @Bean
    @ServiceActivator(inputChannel = "intChannel")
    fun numericPrintService(): PrintService = NumericPrintService(logAdapter())

    @Bean
    @ServiceActivator(inputChannel = "stringChannel")
    fun stringPrintService(): PrintService = StringPrintService(logAdapter())

    @Bean
    fun inputChannel(): MessageChannel = DirectChannel()

    @Bean
    fun stringChannel(): MessageChannel = DirectChannel()

    @Bean
    fun intChannel(): MessageChannel = DirectChannel()

    @Bean
    fun logAdapter(): LogAdapter = CommandLineLogAdapter()
}